# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/BytecodeControlStructures.h
# Commit: 19f956b37c6c
# Full Hash: 19f956b37c6cb832b41b8d1ffb8648f38336f4dd
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-12-19 21:52:02
# Description:
#   Bug 1602392 - Change for-of bytecode to not have an extra value on the stack across the loop edge. r=arai
#   
#   This fixes a bug with HasLiveStackValueAtDepth because now all stack values that
#   are part of the loop (2 values instead of 3 values) are loop-invariant.
#   
# ==============================================================================

diff -r f1547816cf32 -r 19f956b37c6c js/src/frontend/BytecodeControlStructures.h
--- a/js/src/frontend/BytecodeControlStructures.h	Thu Dec 19 10:57:13 2019 +0000
+++ b/js/src/frontend/BytecodeControlStructures.h	Thu Dec 19 09:06:01 2019 +0000
@@ -125,9 +125,6 @@
 
   MOZ_MUST_USE bool emitContinueTarget(BytecodeEmitter* bce);
 
-  // Emit a jump to break target from the top level of the loop.
-  MOZ_MUST_USE bool emitSpecialBreakForDone(BytecodeEmitter* bce);
-
   // `nextPos` is the offset in the source code for the character that
   // corresponds to the next instruction after JSOP_LOOPHEAD.
   // Can be Nothing() if not available.